File: apps/viewer/src/graph/ids.ts
export function stateSubject(fsmId: string, stateId: string): string {
  return `state:${fsmLocalName(fsmId)}.${stateId}`;
}

export function transitionSubject(fsmId: string, transitionId: string): string {
  return `transition:${fsmLocalName(fsmId)}.${transitionId}`;
}

export function eventSubject(fsmId: string, eventId: string): string {
  return `event:${fsmLocalName(fsmId)}.${eventId}`;
}

export function layoutStateId(fsmId: string, stateId: string): string {
  return `layout:${fsmLocalName(fsmId)}.state.${stateId}`;
}

export function layoutTransitionId(fsmId: string, transitionId: string): string {
  return `layout:${fsmLocalName(fsmId)}.transition.${transitionId}`;
}

function fsmLocalName(fsmId: string): string {
  return fsmId.startsWith("fsm:") ? fsmId.slice(4) : fsmId;
}

File: crates/dslraid-codegen/src/view/ids.rs
use dslraid_core::{fsm_local_name, Fsm};

pub(crate) fn layout_state_id(fsm: &Fsm, state: &str) -> String {
    format!("layout:{}.state.{}", fsm_local_name(&fsm.id), state)
}

pub(crate) fn layout_transition_id(fsm: &Fsm, transition: &str) -> String {
    format!(
        "layout:{}.transition.{}",
        fsm_local_name(&fsm.id),
        transition
    )
}
